Fix startup error with some MySQL API compatible databases.

ideal opened the pull request:
e.g. https://github.com/datafuselabs/databend which does not have connection_id().

## Description
Some MySQL compatible databases, may not implemented connection_id() function, on startup, mycli will give error and exit:
```
(1105, 'Code: 2602, displayText = Unknown Function connection_id (while in analyze select projection).')
```

But this is not a fatal error, so we can catch it and continue.

## Checklist
<!--- We appreciate your help and want to give you credit. Please take a moment to put an `x` in the boxes below as you complete them. -->
- [x] I've added this contribution to the `changelog.md`.
- [x] I've added my name to the `AUTHORS` file (or it's already there).

rolandwalker commented:
Hi!  Shouldn't `self.connection_id` be set to some kind of stub value in the exception?

ideal commented:
> Hi! Shouldn't `self.connection_id` be set to some kind of stub value in the exception?

Haven't found an appropriate one, which do you suggest? Thanks.

rolandwalker commented:
@ideal if you `git rebase main` from this branch, the CI issue should be fixed.

ideal commented:
> @ideal if you `git rebase main` from this branch, the CI issue should be fixed.

OK, thanks.

rolandwalker commented:
@ideal it seems like you might have cherry-picked my "pin cryptography" commit instead of doing `git rebase main` which is what is needed to make a clean PR.

Also needed to merge:
 * add this contribution to `changelog.md`
 * add your name to the AUTHORS file

ideal commented:
> @ideal it seems like you might have cherry-picked my "pin cryptography" commit instead of doing `git rebase main` which is what is needed to make a clean PR.
> 

@rolandwalker Sorry, seems that I rebased on wrong remote.

> 
> * add this contribution to `changelog.md`
> * add your name to the AUTHORS file

Updated, thanks.

rolandwalker commented:
Thanks for this nice contribution.  We should do a release soon.
